libraries/AP_GPS/AP_GPS.cpp

```diff
@@ -574,7 +604,8 @@ AP_GPS::update_instance(uint8_t instance)
         return;
     }
 
-    if (drivers[instance] == nullptr || state[instance].status == NO_GPS) {
+    // If it is UAVCAN GNSS - give it regular time as if it is there to acknowledge itself
+    if (drivers[instance] == nullptr || (state[instance].status == NO_GPS && _type[instance] != GPS_TYPE_UAVCAN)) {
```

Comment:
To prevent finding UAVCAN GPS when it actually was not found. This one to save the logic for other types of GPS